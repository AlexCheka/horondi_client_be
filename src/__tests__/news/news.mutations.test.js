/* eslint-disable no-undef */
const { gql } = require('apollo-boost');
const client = require('../../utils/apolloClient');
const { newsMutation } = require('../../modules/news/news.resolver');
require('dotenv').config();
const newsService = require('../../modules/news/news.service');

let newsToUpdateAndDeleteId = '';

describe('news mutations', () => {
  test('add news', async () => {
    const res = await client.mutate({
      mutation: gql`
        mutation {
          addNews(
            news: {
              title: [
                { lang: "uk", value: "тест" }
                { lang: "eng", value: "test" }
              ]
              text: [
                { lang: "ua", value: "тест новина" }
                { lang: "eng", value: "test news" }
              ]
              images: {
                primary: { medium: "sdfsdf4.jpg" }
                additional: { small: "dfgfdg.jpg" }
              }
              video: "3ffefefds.jpg"
              date: "1212121"
            }
          ) {
            _id
          }
        }
      `,
    });

    newsToUpdateAndDeleteId = res.data.addNews._id;
    expect(
      newsMutation.addNews(null, {
        title: [
          { lang: 'uk', value: 'тест' },
          { lang: 'eng', value: 'test' },
        ],
        text: [
          { lang: 'ua', value: 'тест новина' },
          { lang: 'eng', value: 'test news' },
        ],
        images: {
          primary: { medium: 'sdfsdf4.jpg' },
          additional: { small: 'dfgfdg.jpg' },
        },
        video: '3ffefefds.jpg',
        date: '1212121',
      }),
    ).resolves.toBe(res);
    expect(
      newsService.addNews({
        title: [
          { lang: 'uk', value: 'тест' },
          { lang: 'eng', value: 'test' },
        ],
        text: [
          { lang: 'ua', value: 'тест новина' },
          { lang: 'eng', value: 'test news' },
        ],
        images: {
          primary: { medium: 'sdfsdf4.jpg' },
          additional: { small: 'dfgfdg.jpg' },
        },
        video: '3ffefefds.jpg',
        date: '1212121',
      }),
    ).resolves.toBe(res);
  });

  test('update news', async () => {
    const res = await client.mutate({
      mutation: gql`
        mutation {
          updateNews(id:${newsToUpdateAndDeleteId})(
            news: {
              title: [
                { lang: "uk", value: "змінена новина" }
                { lang: "eng", value: "updated test" }
              ]
          ) {
            _id
          }
        }
      `,
    });

    expect(
      newsMutation.addNews(null, {
        title: [
          { lang: 'uk', value: 'тест' },
          { lang: 'eng', value: 'test' },
        ],
        text: [
          { lang: 'ua', value: 'тест новина' },
          { lang: 'eng', value: 'test news' },
        ],
        images: {
          primary: { medium: 'sdfsdf4.jpg' },
          additional: { small: 'dfgfdg.jpg' },
        },
        video: '3ffefefds.jpg',
        date: '1212121',
      }),
    ).resolves.toBe(res);
    expect(
      newsService.addNews({
        title: [
          { lang: 'uk', value: 'тест' },
          { lang: 'eng', value: 'test' },
        ],
        text: [
          { lang: 'ua', value: 'тест новина' },
          { lang: 'eng', value: 'test news' },
        ],
        images: {
          primary: { medium: 'sdfsdf4.jpg' },
          additional: { small: 'dfgfdg.jpg' },
        },
        video: '3ffefefds.jpg',
        date: '1212121',
      }),
    ).resolves.toBe(res);
  });
});
