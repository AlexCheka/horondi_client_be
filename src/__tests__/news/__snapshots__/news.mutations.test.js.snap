// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`news mutations delete news 1`] = `
Object {
  "data": Object {
    "deleteNews": Object {
      "title": Array [
        Object {
          "lang": "uk",
          "value": "тест",
        },
        Object {
          "lang": "eng",
          "value": "blabla",
        },
      ],
    },
  },
}
`;
